src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+import command.EventCommand;
+import command.ExitCommand;
+import command.FindCommand;
+import command.HelpCommand;
+import command.ListCommand;
+import command.ShowAfterCommand;
+import command.ShowBeforeCommand;
+import command.TodoCommand;
+import command.WrongCommand;
+import exception.DescriptionException;
+import exception.DukeDateTimeParserException;
+import exception.DukeKeywordException;
+import exception.NoIndexException;
+import task.DeadlineTask;
+import task.EventTask;
+import task.Task;
+import task.TodoTask;
+
+
+/**
+ * The Parser class are used to parse any user input and process
+ * them into executable command and data.
+ */
+public class Parser {
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    /**
+     * An empty constructor to initialize Parser object.
+     */
+    public Parser() {}
+
+    /**
+     * This method parse a file path into parent directory and the file name.
+     *
+     * @param filepath String The path of a file
+     * @return String[] array of separated String.
+     */
+    public static String[] fileParser(String filepath) {
+        return filepath.split("/");
+    }
+
+
+    /**
+     * This method parse the content of a file line by line and
+     * turns into the correct Task object.
+     *
+     * @param line String Each line of strings in the file.
+     * @return Task Return the correct Task type from reading the file.
+     */
+    public static Task readFileParser(String line) {
+        String[] strings = line.split("\\|", 3);
+        String taskType = strings[0].trim();
+
+        switch(taskType) {
+        case "T" :
+            TodoTask todoTask = new TodoTask();
+            if (strings[1].trim().equals("1")) {
+                todoTask.setDone(true);
+            } else {
+                todoTask.setDone(false);
+            }
+            todoTask.setDescription(strings[2].trim());
+            return todoTask;
+        case "D" :
+            DeadlineTask deadlineTask = new DeadlineTask();
+            if (strings[1].trim().equals("1")) {
+                deadlineTask.setDone(true);
+            } else {
+                deadlineTask.setDone(false);
+            }
+            String[] taskDetails = strings[2].split("\\|");
+            deadlineTask.setDescription(taskDetails[0].trim());
+            deadlineTask.setDate(LocalDateTime.parse(taskDetails[1].trim(), formatter));
+            return deadlineTask;
+        case "E" :
+            EventTask eventTask = new EventTask();
+
+            if (strings[1].trim().equals("1")) {
+                eventTask.setDone(true);
+
+            } else {
+                eventTask.setDone(false);
+            }
+            String[] taskDetails2 = strings[2].split("\\|");
+            eventTask.setDescription(taskDetails2[0].trim());
+            eventTask.setDateTime(LocalDateTime.parse(taskDetails2[1].trim(), formatter));
+            return eventTask;
+        default :
```

Comment:
Code seems to be repeated/very similar. Have you considered abstracting it into a method?